### ISSUE TITLE
v.0.3.0 test fail on power9

### ISSUE THREAD

**Issue by omula** [xianyi/OpenBLAS#1571]
I have tried to compile latest version on power9 and it failed on one of the tests.
TEST 17/22 rot:csrot_inc_0 [FAIL]
  ERR: test_rot.c:109  expected -2.148e-01, got 3.125e-01 (diff -5.273e-01, tol 1.000e-04)

I have launched the compilation with: make -j 160 TARGET=POWER8 BINARY=64 USE_MASS=1
I modified Makefile.power so that -mcpu and -mtune equals to power9.
MASSPATH        = /opt/ibm/xlmass/8.1.6/lib
EXTRALIB       += -L$(MASSPATH) -lmass -lmassvp9 -lmass_simdp9

**Comment by martin-frbg** [comment]
Weird. I had understood from #1469 that this test passes on power8. Can you check if setting 
`CSROTKERNEL=zrot.c` in kernel/power/KERNEL.POWER8 solves it ? (I think it should, as the code path for incx=0 there is the same as in the generic implementation that is known to work. Probably the build system is picking up the old zrot.S instead).

**Comment by quickwritereader** [comment]
try uncomment line @martin-frbg pointed.

One  could add crot.c  based on srot.c if srot passes test. As crot and srot mikrokernels are same and just need c implementation changes. I will try to do it tomorrow.

(Afaik power8 single precision  kernels issue is still open. I am thinking to start that next month and at the same time I am intending to improve initial double prec kernels.)

**Comment by omula** [comment]
It is still failing with the same error after adding CSROTKERNEL=zrot.c in kernel/power/KERNEL.POWER8.

**Comment by martin-frbg** [comment]
Did you do a full rebuild after `make clean` ?

**Comment by omula** [comment]
Yes, I verfied twice.

**Comment by quickwritereader** [comment]
Generic one is 
CROTKERNEL   = ../arm/zrot.c

**Comment by martin-frbg** [comment]
Wait, did i write CSROTKERNEL above :frowning:  ?? 
CROTKERNEL was what I meant (and if you compare the inc=0 section of zrot.c to the arm/zrot.c they are identical, hence my comment above that the c code is known to work.

**Comment by omula** [comment]
Now it worked :)

**Comment by quickwritereader** [comment]
frankly I just searched and could not find on which place it compares incx and incy against 0.

**Comment by martin-frbg** [comment]
It doesn't - I should have written the "not inc=1 section" perhaps (where inc=1 gets optimized with assembly kernel, but the "all other values of inc_x,inc_y" section is identical in the two files). Basically what the utest checks is that the code does __not__ do an immediate exit when there is "nothing to do"

**Comment by quickwritereader** [comment]
Yes it falls back. Tomorrow if I have time I will replace it with modified srot. but for rest power8 single precision I am intending to start next month 
offtopic: @martin-frbg  are you professor of mathematics or data scientist ?

**Comment by martin-frbg** [comment]
offtopic - not a professor and certainly not mathematics. Computational chemistry degree and years of experience reading and writing ugly code that mostly gets the job done.

**Comment by quickwritereader** [comment]
I just saw University on you profile. Well you must still be very good at math, modelling (ml) am I right? I wanted to seek advice from you at those subjects as nowadays it became mainstream.

(Though I  myself got bachelor degree at economics cybernetics but because of bad university curriculum I never became even midlevel at it and thats why instead followed software development which I started from school.)
